src/local_scheduler/local_scheduler.cc

```diff
@@ -210,13 +215,14 @@ void LocalSchedulerState_free(LocalSchedulerState *state) {
   SchedulingAlgorithmState_free(state->algorithm_state);
   state->algorithm_state = NULL;
 
-  /* Destroy the event loop. */
-  destroy_outstanding_callbacks(state->loop);
-  event_loop_destroy(state->loop);
-  state->loop = NULL;
+  event_loop *loop = state->loop;
 
   /* Free the scheduler state. */
   delete state;
+
+  /* Destroy the event loop. */
+  destroy_outstanding_callbacks(loop);
```

Comment:
Why do you re-order these statements with delete?

For a giant PR -- and especially for large PRs -- I think it's best to keep as close to existing behaviors as possible.

Reply:
Yes, this was necessary, since the event loop needs to be destroyed AFTER the AsyncGCSClient is being torn down (which happens in the delete state).